src/sensai/evaluation/eval_stats/eval_stats_classification.py

```diff
@@ -54,8 +50,12 @@ def _computeValue(self, y_true, y_predicted, y_predictedClassProbabilities):
 
 
 class ClassificationMetricTopNAccuracy(ClassificationMetric):
+    requiresProbabilities = True
+
     def __init__(self, n: int):
-        super().__init__(f"Top{n}Accuracy", requiresProbabilities=True)
+        # TODO or not TODO: this is the only metric where the name is not static. We could make a static name like
+        #   "TopNAccuracy" without losing much information
```

Comment:
We would lose _vital_ information. We sometimes use several instances of this with different values of n, and we obviously need to be able to differentiate them when we get our results back.
I don't understand why you would change this in the first place. You are reducing flexibility rather than increasing it (for no apparent reason); this dualism of sometimes static sometimes not is quite strange. There is no reason not to have the string attribute in every instance, really.

Reply:
See comment above, I agree that the attribute should be present in every instance. I currently don't see an elegant solution for this case (and others similar to it). If we can't come up with one, all is left is to revert this commit

Reply:
Will be resolved in the other discussion